packages/@magic-ext/oauth2/src/index.ts

```diff
@@ -64,6 +65,90 @@ export class OAuthExtension extends Extension.Internal<'oauth2'> {
 
     return getResult.call(this, queryString, lifespan);
   }
+
+  public loginWithPopup(configuration: OAuthPopupConfiguration) {
+    let popup: Window | null = null;
+    const width = 448;
+    const height = 568;
+    const left = window.screenLeft + (window.outerWidth / 2 - width / 2);
+    const top = window.screenTop + window.outerHeight * 0.15;
+
+    return this.utils.createPromiEvent<OAuthRedirectResult>(async (resolve, reject) => {
+      // Reject if popup already exists to prevent multiple instances
+      if (popup) {
+        reject(new Error('Popup window already exists'));
+        return;
+      }
+
+      const parseRedirectResult = this.utils.createJsonRpcRequestPayload(OAuthPayloadMethods.Start, [
+        {
+          ...configuration,
+          redirectURI: 'http://localhost:3024/oauth2/popup/verify',
```

Comment:
Do we still need this redirectURI?

Reply:
yes, this URI will be loaded inside of child window to send back the success params

Reply:
Jerry I have changed the implementation according to how we discussed it, now I am passing an extra property - `loginWithPopup` and then assigning redirectURI in iframe i.e. mandrake in main window